src/ImageRunModal.jsx

```diff
@@ -342,7 +342,24 @@ export class ImageRunModal extends React.Component {
         });
     };
 
+    buildFilterRegex(searchText) {
+        // Strip out all non-allowed container image characters when filtering.
+        let regexString = searchText.replace(/[^/\w_.:-]/g, "");
+        // Strip image registry option if set for comparing results for docker.io searching for docker.io/fedora
+        // returns docker.io/$username/fedora for example.
+        if (regexString.includes('/')) {
+            regexString = searchText.replace(searchText.split('/')[0], '');
+        }
```

Comment:
I know this is moved code, but the comment doesn't match up what the code does:
```
> searchText = 'docker.io/user/image:tag'
'docker.io/user/image:tag'
> searchText.replace(searchText.split('/')[0], '')
'/user/image:tag'
```

I.e. it does split off the registry, but the "returns docker.io" comment is wrong. Moreover, that's a weird way of doing that. It relies on `.replace()` changing only the first match by default, but that's by no means intuitive.

I'd find it more readable/explicit to do

    searchText.split('/').slice(1).join('/')

(and possibly prepend another `/` if the search API actually cares for that).

This refactoring could be split out into a separate commit.

Also, I don't understand the comment: "searching for docker.io/fedora" may give you some user/fedora results via the dockerhub search API, but how is that related to `buildFilterRegex()`?

Finally, what's the rationale for stripping off the registry in the first place? Is that a requirement for the search API?

Reply:
This regex is only used for filtering search results and not for the searching itself. When you specify `docker.io/fedora` it will only search docker.io as expected.

What the comment was probably trying to say is when you search for `docker.io/fedora` the results looks like `docker.io/organization/fedora` and using regex `docker.io/fedora` would not match. So either drop the registry from regex or change regex to work with any organization

Reply:
OK -- this is higher-order hacking and I don't know the background of this, so I'll let it slide :grin: